sycl/include/sycl/ext/intel/experimental/esimd/detail/type_format.hpp

```diff
@@ -0,0 +1,84 @@
+//==-------------- types.hpp - DPC++ Explicit SIMD API ---------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// Meta-functions to compute compile-time element type of a simd_view resulting
+// from format operations.
+//===----------------------------------------------------------------------===//
+
```

Comment:
Content of this file was moved from types.hpp, mostly verbatim. The changes are noted below.